### PR TITLE
handle onNewIntent in ActivityFlowSupport

### PR DESCRIPTION
**Pull request by loganj**

### CODE REVIEW

File: flow/src/main/java/flow/ActivityFlowSupport.java [hunk 1]
```diff
@@ -68,48 +69,61 @@ public NonConfigurationInstance(Flow flow) {
...
     }
-    return new ActivityFlowSupport(new Flow(initialBackstack), parceler);
+    return new ActivityFlowSupport(flow, parceler, intent);
   }
 
-  public NonConfigurationInstance onRetainNonConfigurationInstance() {
-    return new NonConfigurationInstance(flow);
+  public void onNewIntent(Intent intent) {
+    checkArgument(intent != null, "intent may not be null");
+    this.newIntent = intent;
```
Comment (rjrjr) [inline flow/src/main/java/flow/ActivityFlowSupport.java]: Need to verify that this is always followed by onResume

**Comment by loganj** [comment]
Verified the `onResume()` behavior, but also moved the intent handling around so it's simpler and clearer (and verified the behavior again).

File: flow-sample/src/main/AndroidManifest.xml [hunk 2]
```diff
@@ -28,11 +28,11 @@
     <activity
         android:label="Flow"
         android:name=".MainActivity"
+        android:launchMode="singleTop"
```
Comment (rjrjr) [inline flow-sample/src/main/AndroidManifest.xml]: singleInstance?

File: flow/src/main/java/flow/ActivityFlowSupport.java [hunk 3]
```diff
@@ -68,10 +69,15 @@ public NonConfigurationInstance(Flow flow) {
     }
   }
 
-  private static final String FLOW_KEY = ActivityFlowSupport.class.getSimpleName() + "_backstack";
+  public static void setBackstackExtra(Intent intent, Backstack backstack, Parceler parceler) {
+    intent.putExtra(BACKSTACK_KEY, backstack.getParcelable(parceler));
```
Comment (rjrjr) [inline flow/src/main/java/flow/ActivityFlowSupport.java]: You're making it impossible to manage more than one flow with this static key. Should we allow (not require) specifying a custom key?

File: flow/src/main/java/flow/ActivityFlowSupport.java [hunk 4]
```diff
@@ -80,36 +86,42 @@ private ActivityFlowSupport(Flow flow, Parceler parceler) {
...
+      Intent intent, Bundle savedInstanceState, Parceler parceler, Backstack defaultBackstack) {
     checkArgument(parceler != null, "parceler may not be null");
-    checkArgument(defaultBackstack != null, "defaultBackstack may not be null");
-    Backstack initialBackstack = null;
-    if (savedInstanceState != null && savedInstanceState.containsKey(FLOW_KEY)) {
-      initialBackstack = Backstack.from(savedInstanceState.getParcelable(FLOW_KEY), parceler);
-    }
-    if (initialBackstack == null) {
-      initialBackstack = defaultBackstack;
+    final Flow flow;
```
Comment (rjrjr) [inline flow/src/main/java/flow/ActivityFlowSupport.java]: final not needed

**Comment by rjrjr** [comment]
Two nits, merging anyway.

**Pull request merged**